PhysicsTools/NanoAOD/python/muons_cff.py

```diff
@@ -31,6 +31,32 @@
     srcVtx = cms.InputTag("offlineSlimmedPrimaryVertices"),
 )
 
+muonMVAID = cms.EDProducer("EvaluateMuonMVAID",
+    src = cms.InputTag("slimmedMuonsUpdated"),
+    weightFile =  cms.FileInPath("RecoMuon/MuonIdentification/data/mvaID.onnx"),
+    isClassifier = cms.bool(False),
+    backend = cms.string('ONNX'),
+    name = cms.string("muonMVAID"),
+    outputTensorName= cms.string("probabilities"),
+    inputTensorName= cms.string("float_input"),
+    outputNames = cms.vstring(["probGOOD", "wpMedium", "wpTight"]),
+    batch_eval =cms.bool(True),
+    outputFormulas = cms.vstring(["at(1)", "? at(1) > 0.14 ? 1 : 0", "? at(1) > 0.35 ? 1 : 0"]),
```

Comment:
> The finalMuons are [selected using an OR of a few IDs](https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/muons_cff.py#L50). Are you sure you don't want to keep muon passing the medium mvaID WP? (said otherwise, is it possible to fail that cut but still pass the medium mvaID WP?)

The finalMuon selection is much looser than the medium WP of the mvaID. Also, muons selected by the mvaID wp medium have to pass looseID, hence those are finalMuons.

Reply:
Thanks for clarifying that point!